### ISSUE TITLE
[ENHANCEMENT] Add pins on click with info boxes

### ISSUE THREAD

**Issue by lebster** [ondrabus/kentico-mvcwidget-bing-maps#1]
@ondrabus ,

This is a good example widget. Maybe as an advanced version of this widget it would be nice if the content editors can add multiple pins. As well as adding pins maybe they can add info box data (Title, Description) for each pin added.

Is it a good idea to store Json into a widget text property? We could store the pin locations/info boxs in json and save it into a single widget property. This way in page builder they can add onclick.

If that is a bad idea, what would be the best way to add this feature?

**Comment by ondrabus** [comment]
Hi @lebster,
generally you should not store any content in widget properties. Widget is code - implementation - while the JSON you mentioned is pure content. Therefore in this case I'd store the data for pins in pages (structured content based on defined page type) nested below the page containing the widget. That way the widget can look at content below it's actual position in the content tree without any additional configuration.
In other words, the widget configuration should only contain configuration data, not content.
Hope this makes sense :-)
